### PR TITLE
fix: high resources panic

### LINKED ISSUE #3841: I got a panic error when executing `okteto up` and my `resources.requests` are too high
**Describe the bug**

I had deployed the [`go-getting-started`](https://github.com/okteto/go-getting-started) example and modified the okteto manifest to set some requests to me dev container and when I executed `okteto up` I'm getting a panic error:

```
okteto up
 i  Using ifbyol @ xxxxxxxxx as context
 i  'go-getting-started' was already deployed. To redeploy run 'okteto deploy' or 'okteto up --deploy'
 i  Images were already built. To rebuild your images run 'okteto build' or 'okteto deploy --build'
panic: runtime error: index out of range [1] with length 0

goroutine 87 [running]:
github.com/okteto/okteto/pkg/k8s/deployments.getResourceLimitError({0xc0001d2a80, 0x75}, 0xc001d52600)
	github.com/okteto/okteto/pkg/k8s/deployments/crud.go:186 +0x3f3
github.com/okteto/okteto/pkg/k8s/deployments.CheckConditionErrors(0x3c67f30?, 0xc001c74480?)
	github.com/okteto/okteto/pkg/k8s/deployments/crud.go:158 +0x331
github.com/okteto/okteto/pkg/k8s/apps.(*DeploymentApp).CheckConditionErrors(0xc001c64198?, 0x3c43bb0?)
	github.com/okteto/okteto/pkg/k8s/apps/deployments.go:111 +0x1d
github.com/okteto/okteto/pkg/k8s/apps.GetRunningPodInLoop({0x3c43bb0, 0xc0000b0690}, 0xc001d52600, {0x3c56448, 0xc001c64198}, {0x3c67f30, 0xc001c74480})
	github.com/okteto/okteto/pkg/k8s/apps/crud.go:73 +0x107
github.com/okteto/okteto/cmd/up.(*upContext).createDevContainer(0xc0000de160, {0x3c43bb0, 0xc0000b0690}, {0x3c56448, 0xc000193e00}, 0x0)
	github.com/okteto/okteto/cmd/up/activate.go:308 +0x76d
github.com/okteto/okteto/cmd/up.(*upContext).devMode(0xc0000de160?, {0x3c43bb0, 0xc0000b0690}, {0x3c56448, 0xc000193e00}, 0x6?)
	github.com/okteto/okteto/cmd/up/activate.go:246 +0x32
github.com/okteto/okteto/cmd/up.(*upContext).activate(0xc0000de160)
	github.com/okteto/okteto/cmd/up/activate.go:138 +0xaa5
github.com/okteto/okteto/cmd/up.(*upContext).activateLoop(0xc0000de160)
	github.com/okteto/okteto/cmd/up/up.go:705 +0x296
created by github.com/okteto/okteto/cmd/up.(*upContext).start
	github.com/okteto/okteto/cmd/up/up.go:637 +0x6e5
```

Checking the stack trace, it seems we are trying to access to an element of an empty array. It seems the error is because the memory I specified is too high. I wasn't able to reproduce it with other examples like movies.

**To Reproduce**

1. Deploy go-getting-started
2. Modify the okteto manifest to specify the following: `resources.requests.cpu`=`200m` and `resources.requests.memory`=`10Gi`
3. Execute `okteto up`

**Expected behavior**

I shouldn't get a panic. If the operation cannot be done for whatever reason I should get a proper error message

**Desktop (please complete the following information):**

- Version: `2.18.0`

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by jLopezbarb**
# Proposed changes

Fixes #3841 

Fix a panic that was happening when the message produced by k8s was missing any resource and limit. Given that I was not able to reproduce it, I added tests covering the use cases where the error could happen based on the logs provided. 

## How to validate

Check unit tests. No way to reproduce the issue on any cluster. I just could reproduce it on unit tests.
 
## CLI Quality Reminders 🔧

For both authors and reviewers:

- Scrutinize for potential regressions
- Ensure key automated tests are in place
- Build the CLI and test using the validation steps
- Assess Developer Experience impact (log messages, performances, etc)
- If too broad, consider breaking into smaller PRs
- Adhere to our [code style](https://github.com/okteto/okteto/blob/master/docs/code-style.md) and [code review](https://github.com/okteto/okteto/blob/master/docs/code-review.md) guidelines

### CODE REVIEW

File: pkg/k8s/deployments/crud.go [hunk 1]
```diff
@@ -174,21 +174,31 @@ func getResourceLimitError(errorMessage string, dev *model.Dev) error {
 	var errorToReturn string
 	if strings.Contains(errorMessage, "maximum cpu usage") {
 		cpuMaximumRegex := regexp.MustCompile(`cpu usage per Pod is (\d*\w*)`)
-		maximumCpuPerPod := cpuMaximumRegex.FindStringSubmatch(errorMessage)[1]
-		var manifestCpu string
-		if limitCpu, ok := dev.Resources.Limits[apiv1.ResourceCPU]; ok {
-			manifestCpu = limitCpu.String()
+		maximumCpuPerPodMatchGroups := cpuMaximumRegex.FindStringSubmatch(errorMessage)
+		if len(maximumCpuPerPodMatchGroups) < 2 {
```
Comment (teresaromero) [inline pkg/k8s/deployments/crud.go:178]: Why 2, can we put this value under a const with name?

File: pkg/k8s/deployments/crud.go [hunk 2]
```diff
@@ -174,21 +174,31 @@ func getResourceLimitError(errorMessage string, dev *model.Dev) error {
...
-		errorToReturn += fmt.Sprintf("The value of resources.limits.cpu in your okteto manifest (%s) exceeds the maximum CPU limit per pod (%s). ", manifestCpu, maximumCpuPerPod)
 	}
 	if strings.Contains(errorMessage, "maximum memory usage") {
 		memoryMaximumRegex := regexp.MustCompile(`memory usage per Pod is (\d*\w*)`)
-		maximumMemoryPerPod := memoryMaximumRegex.FindStringSubmatch(errorMessage)[1]
-		var manifestMemory string
-		if limitMemory, ok := dev.Resources.Limits[apiv1.ResourceMemory]; ok {
-			manifestMemory = limitMemory.String()
+		maximumMemoryPerPodMatchGroups := memoryMaximumRegex.FindStringSubmatch(errorMessage)
+		if len(maximumMemoryPerPodMatchGroups) < 2 {
```
Comment (teresaromero) [inline pkg/k8s/deployments/crud.go:192]: same :)

**Review by teresaromero — APPROVED** [review]
nit: just comment on a value, overrall LGTM

**Review by AdrianPedriza — APPROVED** [review]

File: pkg/k8s/deployments/crud_test.go [hunk 3]
```diff
@@ -334,19 +390,91 @@ func TestCheckConditionErrors(t *testing.T) {
...
-			if err == nil {
-				t.Fatalf("Didn't receive any error. Expected %s", tt.expectedErr)
+			if err == nil && tt.expectedErr != nil {
+				t.Fatalf("wrong error. Got nil, expected %s", tt.expectedErr.Error())
 			}
-
-			if err.Error() != tt.expectedErr.Error() {
-				t.Fatalf("wrong error. Got %s, expected %s", err, tt.expectedErr)
+			if err != nil && tt.expectedErr == nil {
+				t.Fatalf("wrong error. Got nil, expected %s", err.Error())
```
Comment (andreafalzetti) [inline pkg/k8s/deployments/crud_test.go:477]:
```suggestion
				t.Fatalf("wrong error. Expected nil, but got %s", err.Error())
```

**Review by andreafalzetti — APPROVED** [review]

**Pull request merged**